aten/src/TH/generic/THBlas.cpp

```diff
@@ -381,52 +424,71 @@ void THBlas_(gemm)(char transa, char transb, int64_t m, int64_t n, int64_t k, sc
           for(l = 0; l < k; l++)
             sum += a_[l]*b_[l];
           b_ += ldb;
-	  if (beta == 0)
-	    c[j*ldc+i] = alpha*sum;
-	  else
-	    c[j*ldc+i] = beta*c[j*ldc+i]+alpha*sum;
+          if (beta == 0)
+            c[j*ldc+i] = alpha*sum;
+          else
+            c[j*ldc+i] = beta*c[j*ldc+i]+alpha*sum;
         }
         a_ += lda;
       }
     }
     else if(!transa_ && transb_)
     {
-      scalar_t *a_ = a;
-      for(i = 0; i < m; i++)
-      {
-        scalar_t *b_ = b;
-        for(j = 0; j < n; j++)
-        {
-          scalar_t sum = 0;
-          for(l = 0; l < k; l++)
-            sum += a_[l*lda]*b_[l*ldb];
-          b_++;
-	  if (beta == 0)
-	    c[j*ldc+i] = alpha*sum;
-	  else
-	    c[j*ldc+i] = beta*c[j*ldc+i]+alpha*sum;
+      if (beta == 0) {
```

Comment:
Do we explicitly test for beta == 0 and beta != 0? It's important to confirm this.

Reply:
for the caffe2/test under fbcode (i.e. "buck test @mode/dev caffe2/test:torch"), the value of beta is set to zero.

Reply:
Ok, can you look for callsites of THBlas_(gemm) to see which ones might call it with beta != 0? I can also help you with this one-on-one if necessary.